Rebase onto .NET Core 1.1

randomvariable opened the pull request:
`Build.psm1` and `project.json` files modified to use .NET Core 1.1.

.NET Core 1.1 ships with an older dotnet/cli than has currently been used, 
so we revert to use case-sensitive directory names for dependencies.

.NET Core 1.1 is a pre-requisite for supporting Fedora 24.

msftclas commented:
Hi __@randomvariable__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, MSBOT;

vors commented:
> .NET Core 1.1 ships with an older dotnet/cli than has currently been used

That should not really matter, although, I think it could be acceptable to revert back for consistency.
Since we are not using nightly `dotnet` cli anymore.
Last pinned version of `dotnet` cli was completely random and it's probably better to stick with the one that corresponds to some release.

randomvariable commented on docs/building/windows-core.md:
```diff
@@ -24,7 +24,7 @@ This guide assumes that you have recursively cloned the PowerShell repository an
 Visual Studio
 ----------------
 
-You will need to install an edition of Visual Studio 2015 (Community, Enterprise, or Professional) with the optional feature 'Common Tools for Visual C++' installed. 
```
Thanks. Missed when diffing.

randomvariable commented:
Appears the build in #2738 that I've got out of that doesn't allocate a tty, so is pretty unusable as an interactive shell. Not sure why.

randomvariable commented:
TTY bug seems unrelated to .NET 1.1, have filed issue at https://github.com/PowerShell/PowerShell/issues/2745

randomvariable commented:
Not sure why the DSC tests are failing. Any ideas?

randomvariable committed:
Rebase onto .NET Core 1.1

Modify `Build.psm1` and `project.json` files to use .NET Core 1.1.

.NET Core 1.1 ships with an older dotnet/cli than has currently been used,
so we revert to use case-sensitive directory names for dependencies.

.NET Core 1.1 is a pre-requisite for supporting Fedora 24.

randomvariable commented:
Think this is ready to merge.

daxian-dbw commented on src/Microsoft.PowerShell.SDK/project.json:
```diff
@@ -14,91 +14,91 @@
         "net451": {},
         "netstandard1.6": {
-            "imports": ["dnxcore50", "portable-net45+win8"],
+            "imports": ["netcoreapp1.1", "portable-net45+win8"],
```
Why do you change `dnxcore50` to `netcoreapp1.1` here?

randomvariable replied:
That's an error. Will revert.

daxian-dbw replied:
Thanks for the quick response. Please add new commits to address comments, it's easier to review the changes this way 😄 I will squash the commits when merging the PR.

daxian-dbw commented on test/PSReadLine/project.json:
```diff
@@ -13,10 +13,10 @@
     },
```
The dependency to `Microsoft.PowerShell.PSReadLine` should be `"6.0.0-*"`. It's an oversight from the change that bounce the project versions. Would you mind update it in this PR?

randomvariable replied:
Done. Commits upcoming.

daxian-dbw commented on src/Microsoft.PowerShell.Commands.Utility/project.json:
```diff
@@ -105,8 +105,8 @@
             "dependencies": {
-                "Microsoft.CodeAnalysis.CSharp": "1.3.1-beta1-20160616-03",
-                "System.Diagnostics.TextWriterTraceListener": "4.3.0-preview1-24530-04"
+                "Microsoft.CodeAnalysis.CSharp": "1.3.2",
```
Since we are using the latest version on nuget.org for other packages, shall we use the [`2.0.0-rc` version](https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp/2.0.0-rc) for `Microsoft.CodeAnalysis.CSharp`? The `2.0.0-rc` was published on Nov 18.

randomvariable replied:
The build and tests succeeded against 2.0.0-rc, so this looks fine to go in.

daxian-dbw reviewed (changes requested):
This is an awesome PR! Thank you very much!
I left a few comments, can you please take a look?

randomvariable commented on build.psm1:
```diff
@@ -924,11 +922,11 @@ function Install-Dotnet {
         Start-NativeExecution {
             curl -sO $obtainUrl/$installScript
-            bash ./$installScript -c $Channel -v $Version
+            bash ./$installScript -c $Channel
```
I think I had removed this whilst trying to get the 1.1 install script working, but the defaults are now set correctly elsewhere in build.psm1, so it can be restored.

randomvariable committed:
PSReadLine: Bump major version of PSReadline to 6.0.0-*

randomvariable committed:
Microsoft.PowerShell.Commands.Utility: Bump major version of Microsoft.CodeAnalysis.Csharp

randomvariable committed:
Microsoft.PowerShell.SDK: Revert attempt to import netcoreapp1.1 instead of dnxcore

randomvariable committed:
build.psm1: Restore ability to specify version of dotnet-install.sh

randomvariable commented:
Commits added for each of the review comments.

daxian-dbw reviewed (approved):
LGTM.
It's my bad that I didn't make it clear -- you can use one commit to address any number of comments. Sorry to put extra work on your side 😄

daxian-dbw commented:
The Travis CI failure is a known issue, not related to your PR. I have restarted it. I will merge this PR once it succeeds.

daxian-dbw commented:
@randomvariable Thanks again for your contribution!
